src/widgets/line/lv_line.c

```diff
@@ -91,6 +89,42 @@ void lv_line_set_y_invert(lv_obj_t * obj, bool en)
  * Getter functions
  *====================*/
 
+const lv_point_precise_t * lv_line_get_points(lv_obj_t * obj)
+{
+    LV_ASSERT_OBJ(obj, MY_CLASS);
+
+    lv_line_t * line = (lv_line_t *)obj;
+    return line->point_array.constant;
+}
+
+uint32_t lv_line_get_point_count(lv_obj_t * obj)
+{
+    LV_ASSERT_OBJ(obj, MY_CLASS);
+
+    lv_line_t * line = (lv_line_t *)obj;
+    return line->point_num;
+}
+
+bool lv_line_get_point_array_is_mutable(lv_obj_t * obj)
+{
+    LV_ASSERT_OBJ(obj, MY_CLASS);
+
+    lv_line_t * line = (lv_line_t *)obj;
+    return line->point_array_is_mutable;
+}
+
+lv_point_precise_t * lv_line_get_points_mutable(lv_obj_t * obj)
+{
+    LV_ASSERT_OBJ(obj, MY_CLASS);
+
+    lv_line_t * line = (lv_line_t *)obj;
+    if(!line->point_array_is_mutable) {
+        LV_LOG_ERROR("the line point array is not mutable");
```

Comment:
It should only a warning. `ERROR`s are rather used on critical problems or crashes.